htdocs/fourn/facture/card.php

```diff
@@ -946,23 +970,46 @@
 				$action = 'create';
 			}
 
+			if (getDolGlobalInt('INVOICE_SUBTYPE_ENABLED') && empty(GETPOST("subtype"))) {
+				$error++;
+				setEventMessages($langs->trans("ErrorFieldRequired", $langs->transnoentitiesnoconv("InvoiceSubtype")), null, 'errors');
+				$action = 'create';
+			}
+
+			if ($mysoc->country_code == 'GR' && getDolGlobalInt('INVOICE_SUBTYPE_ENABLED') && GETPOSTISSET('subtype')) {
+				$subtypearray = $object->subtype_array(1);
+				$subtype = GETPOST('subtype');
+				foreach ($subtypearray as $code => $rowid) {
+					if ($rowid == $subtype) {
+						$selectedcode = $code;
+						break;
+					}
+				}
+				if (in_array($selectedcode, array('5.1', '5.2', '11.4'))) {
```

Comment:
Hard coded value are not allowed. I suggest to remove completely this test for the first version.

Reply:
Fixed
Any suggestion on how i could tackle this because it's about the most common mistake made by users having to choose between so many subtypes and is driving accountants crazy!?

fyi the hardcoded values i used are from the revenue ministry api and never gonna change.